bigchaindb/backend/query.py

```diff
@@ -328,8 +330,28 @@ def get_txids_filtered(connection, asset_id, operation=None):
 
 
 @singledispatch
-def text_search(conn, search, language='english', case_sensitive=False,
+def text_search(conn, search, *, language='english', case_sensitive=False,
                 diacritic_sensitive=False, text_score=False, limit=0):
-    # TODO: docstring
+    """Return all the assets that match the text search.
 
-    raise NotImplementedError
+    The results are sorted by text score.
+
+    Args:
+        search (str): Text search string to query the text index
+        language (str, optional): The language for the search and the rules for
+            stemmer and tokenizer. If the language is `None` text search uses
+            simple tokenization and no stemming.
+        case_sensitive (bool, optional): Enable or disable case sensitive
+            search.
+        diacritic_sensitive (bool, optional): Enable or disable case sensitive
+            diacritic search.
+        text_score (bool, optional): If `True` returns the text score with
+            each document.
+        limit (int, optional): Limit the number of returned documents.
+
+    Returns:
+        :obj:`list` of :obj:`dict`: a list of assets
+    """
+
+    raise OperationError('This query is only supported when running '
+                         'BigchainDB with MongoDB as the backend.')
```

Comment:
Not sure what is the ideal way to document this, but it would be nice to have some pointers, or brief explanations about stemming.

We could perhaps simply point to the MongoDB relevant docs, such as https://docs.mongodb.com/manual/reference/operator/query/text/#stemmed-words

I was a bit puzzled to see results with "baking" returned for the word "bake". The stemming behaviour explains it though:

> For case insensitive and diacritic insensitive text searches, the $text operator matches on the complete stemmed word. So if a document field contains the word blueberry, a search on the term blue will not match. However, blueberry or blueberries will match.

I think that is an important detail.